src/vs/base/browser/ui/countBadge/countBadge.ts

```diff
@@ -57,14 +58,16 @@ export class CountBadge extends Disposable {
 
 	setTitleFormat(titleFormat: string) {
 		this.titleFormat = titleFormat;
+		this.updateHover();
 		this.render();
 	}
 
 	private updateHover(): void {
-		this.hover?.dispose();
-		this.hover = undefined;
-		if (this.titleFormat !== '') {
-			this.hover = getBaseLayerHoverDelegate().setupDelayedHoverAtMouse(this.element, { content: format(this.titleFormat, this.count), appearance: { compact: true } });
+		if (this.titleFormat !== '' && !this.hover) {
+			this.hover = getBaseLayerHoverDelegate().setupDelayedHoverAtMouse(this.element, () => ({ content: format(this.titleFormat, this.count), appearance: { compact: true } }));
+		} else if (this.titleFormat === '' && this.hover) {
+			this.hover?.dispose();
+			this.hover = undefined;
```

Comment:
`MutableDisposable` would make this a little nicer and let you register in the prob instead of overriding dispose